### ISSUE TITLE
Touchpad temp vs measured temp as different entities

### ISSUE THREAD

**Issue by Jemzlel** [nathanvdh/homeassistant-airtouch2#14]
Hi there, love this integration.
One thing that I couldn't seem to find is whether there was a different entity option for those that have configured their AirTouch2 to be touchpad temp.
I have configured mine to be touchpad temp, because when it's cold overnight (or hot in summer) and I have the default setting (i.e. Thermistor on AC = Ticked) in AC SETUP on the AirTouch 2 controller, the temp that's displayed is wildly different to when the setting is Thermistor on AC = Unticked.
The reason I know this is because I believe the temp value in this integration is 20 in Home Assistant, but when I look at the touchpad or my iPhone companion app, it's showing as 24. Which is the temp at the touchpad.
I did read through the PDF and it looks as though there's something for item 323 143 saying Touchpad temp and I saw some lines in the 
src/airtouch2/protocol/at2/messages/SystemInfo.py file.

Was wondering if there's a way to record the two different values as different entities and we can decide which value we want to use.

Also, I wasn't sure how to get the temp value out as an entity, then track it like any other temp sensor. (I'm very new to home assistant)

**Comment by the-inconvenience-store** [comment]
I have [submitted a PR in the airtouch2-python](https://github.com/nathanvdh/airtouch2-python/pull/14) to address this issue. Currently, Touchpad Temp is read from the AC unit, but not saved into the SystemInfo class. Once the PR is merged & package is updated, I'll make another PR here to address this issue & enable choosing between touchpad temp and measured temp during setup. @nathanvdh would you consider adding me as a maintainer/contributor in both these packages?  I've got a few fixes & improvements lined up. Thanks in advance.

**Comment by Jemzlel** [comment]
@the-inconvenience-store  This is awesome, thanks for the help.

When you say choosing during setup, does that mean you could only display/show one or the other.

It would be good to have both as sensors to see the difference, but use one in the main climate card.

Reason being is currently I can't rely on the AC temp, because in my roof in summer it gets up to 50c. Whereas my touchpad temp is something like 25/26. I have other tuya temp sensors around the house which I used instead.

But would be good to have airtouch 2 touchpad temp as a entity/sensor and airtouch 2 AC unit temp as a separate one, which can be separate to the chosen one for the setup, if that makes sense.

Thanks,

**Comment by the-inconvenience-store** [comment]
@Jemzlel, no worries 😸 

> When you say choosing during setup, does that mean you could only display/show one or the other.

Yeah something along those lines. 

> It would be good to have both as sensors to see the difference, but use one in the main climate card.

Should be easy enough. @nathanvdh any thoughts?

Nathan's requested some changes to my PR, but if all goes through swiftly should be up in a few days time. I'll update this issue with my PR when it's up.

**Comment by nathanvdh** [comment]
The climate entity uses the one from the AC info, I'm not immediately sure how you would implement choosing one or the other sensor. I think the entity doesn't even have access to the At2Client where the touchpad temp is at the moment.

https://github.com/nathanvdh/homeassistant-airtouch2/blob/master/custom_components/airtouch2/Airtouch2ClimateEntity.py

Adding another sensor entity that just has the temp sensor in it should be easy though.

I'd be happy with you implementing both things. I'm not going to be doing much more work on this project myself though.

**Comment by the-inconvenience-store** [comment]
> I'm not immediately sure how you would implement choosing one or the other sensor.

The `At2Client` is available through the `At2Aircon` class in airtouch2-python. 

```python
...
class At2Aircon(Publisher):
    info: AcInfo

    def __init__(self, client: At2Client, info: AcInfo):
        self._client: At2Client = client
        self._callbacks: list[Callable] = []
        self.info = info
...
```
Slightly hacky, but gets the job done.

> I'd be happy with you implementing both things. I'm not going to be doing much more work on this project myself though.

No worries, I figured as much. Just glad to see someones at least implemented this ancient thing for HASS.

**Comment by Jemzlel** [comment]
So for the GitHub noob, now that there’s a pull request/change, does that turn into a new release and I will that turn into an “update” in HASS?

**Comment by the-inconvenience-store** [comment]
> So for the GitHub noob, now that there’s a pull request/change, does that turn into a new release and I will that turn into an “update” in HASS?

Soon, @nathanvdh needs to push an update for the airtouch2-python library (the library this integration relies on) to PyPi with my changes. After the update I can put my update for the integration up which you can then update in HASS.

**Comment by Jemzlel** [comment]
Ahk, sweet
